### PR TITLE
Improve lwgeom_geos clarity

### PR DESCRIPTION
**Pull request by dbaston**
This PR attempts to clarify some of the code in `lwgeom_geos.c` by removing function calls that require non-obvious arguments (which are often `NULL` or `__func__`) and replacing them with calls to variadic macros/functions.

### CODE REVIEW

File: liblwgeom/lwgeom_geos.c [hunk 1]
```diff
@@ -495,92 +535,59 @@ lwgeom_geos_version()
...
-}
-
-/* Input decoder and sanity checker for GEOS wrappers */
-inline static uint8_t
-input_lwgeom_to_geos(GEOSGeometry** g, const LWGEOM* geom, const char* funcname)
+/* Return the consistent SRID of all inputs, or issue an
+ * error in case of SRID mismatch. Intended to be called
+ * from RESULT_SRID macro */
+static int32_t
+get_result_srid(int count, const char* funcname, ...)
```
Comment (Algunenano) [inline liblwgeom/lwgeom_geos.c]: Since `sizeof` returns size_t I'd rather have both `count` and `i` as size_t too.

**Comment by pramsey** [comment]
I like it, 👍

**Comment by dbaston** [comment]
@Komzpa do you want to review this or should I merge it in?

File: liblwgeom/lwgeom_geos.c [hunk 2]
```diff
@@ -49,6 +51,44 @@ lwgeom_geos_error(const char* fmt, ...)
 	va_end(ap);
 }
 
+/* Destroy any non-null GEOSGeometry* pointers passed as arguments */
+#define GEOS_FREE(...) \
```
Comment (Komzpa) [inline liblwgeom/lwgeom_geos.c]: is "destroy" = "free"?

**Comment by Komzpa** [comment]
@dbaston I'm ok with merge of this. Only thing I'm at doubt of is scope of current freeze and thus if this should go to 3.0 or 2.5.

**Comment by dbaston** [comment]
Thanks for looking, @Komzpa . I'm inclined to target 2.5 so that we don't have three very different versions of this file in the wild.

**Comment by robe2** [comment]
Okay go ahead :+1:

**Pull request closed without merging**